src/clients/api/queries/getIsolatedPools/formatOutput/formatDailyDistributedRewardTokensMapping.ts

```diff
@@ -0,0 +1,84 @@
+import BigNumber from 'bignumber.js';
+import { ContractCallReturnContext } from 'ethereum-multicall';
+import _cloneDeep from 'lodash/cloneDeep';
+import { Token } from 'types';
+import { calculateDailyDistributedTokens, getTokenByAddress } from 'utilities';
+
+import { logError } from 'context/ErrorLogger';
+
+export interface FormatDailyDistributedRewardTokensMappingInput {
+  tokenPricesDollars: {
+    [tokenAddress: string]: BigNumber;
+  };
+  rewardsDistributorsResults: ContractCallReturnContext[];
+}
+export interface DailyDistributedRewardTokensMapping {
+  [vTokenAddress: string]: {
+    rewardToken: Token;
+    rewardTokenPriceDollars: BigNumber;
+    supplyDailyDistributedRewardTokens: BigNumber;
+    borrowDailyDistributedRewardTokens: BigNumber;
+  }[];
+}
+
+const formatDailyDistributedRewardTokensMapping = ({
+  rewardsDistributorsResults,
+  tokenPricesDollars,
+}: FormatDailyDistributedRewardTokensMappingInput) =>
+  rewardsDistributorsResults.reduce<DailyDistributedRewardTokensMapping>(
+    (acc, rewardsDistributorsResult) => {
+      const results = rewardsDistributorsResult.callsReturnContext;
+      const rewardTokenAddress = results[0].returnValues[0].toLowerCase();
+      const rewardToken = getTokenByAddress(rewardTokenAddress);
+
+      if (!rewardToken) {
+        logError(`Record missing for reward token: ${rewardTokenAddress}`);
+        return acc;
+      }
+
+      const rewardTokenPriceDollars = tokenPricesDollars[rewardToken.address.toLowerCase()];
+
+      if (!rewardTokenPriceDollars) {
+        logError(
+          `Price could not be fetched for reward token: ${rewardToken.symbol} (${rewardToken.address})`,
+        );
+        return acc;
+      }
+
+      const speedResults = results.slice(1);
+      const accCopy = _cloneDeep(acc);
+
+      for (let r = 0; r < speedResults.length - 1; r += 2) {
+        const vTokenAddress = speedResults[r].methodParameters[0].toLowerCase();
+        const supplySpeedMantissa = new BigNumber(speedResults[r].returnValues[0].hex);
+        const borrowSpeedMantissa = new BigNumber(speedResults[r + 1].returnValues[0].hex);
+
+        // Only add distribution if one of the speeds is not 0
+        if (supplySpeedMantissa.isGreaterThan(0) || borrowSpeedMantissa.isGreaterThan(0)) {
+          const supplyDailyDistributedRewardTokens = calculateDailyDistributedTokens({
+            ratePerBlockMantissa: supplySpeedMantissa,
+            decimals: rewardToken.decimals,
```

Comment:
@kkirka @chechu 
Is it correct here to use the decimals of the reward token, and not 18 decimals all the time?
I'm a bit confused with what decimals are used for what values. Manmeet confirmed me the `supplyRatePerBlock` and `borrowRatePerBlock` values of a VToken (returned in the `getAllPools` response of the PoolLens contract for exampe) always use 18 decimals, yet we recently implemented a fix to calculate distribution APYs using the decimals of the reward token.

Reply:
I think the confusion comes from the fact we mix up "rate", "speed", etc. terms.
1) For a value that is inherently a _percentage_ like interest rates (supplyRatePerBlock) or factors (collateral factor, liquidation threshold, etc.) we use 18 decimals.
2) For values that are expressed in _tokens_ (e.g., distribution speed — how many tokens are distributed per block) we use token decimals.
3) For values that are in vTokens (e.g., vToken.totalSupply(), vToken.balanceOf(user)) we use vToken decimals which is 8 for all vTokens currently, and we'll most likely follow this convention going forward.

I'm not sure how we can fix this confusion though. Maybe we have this info missing in docs somewhere? We can add it, can you list the contract functions you think this info should be added to?

Reply:
I see. I remember we had that discussion but I still wasn't sure this was correct. Thanks for the explanation.

Adding this to a doc somewhere would be great yeah: in Notion perhaps?
For now I'm clear, these 3 rules should be enough.

Reply:
We could use custom types for this, e.g.
```
type Exp is uint256; // 18 decimals
type UnderlyingAmount is uint256; // underlying decimals
type VTokenAmount is uint256; // vToken decimals
type Price is uint256; // (36 - underlying decimals) decimals
type ExchangeRate is uint256; // (underlying decimals + 18 - vToken decimals) decimals 
```

...and then have conversion functions in the code.

I experimented with this, but it would require quite a huge refactoring of the codebase, so I figured it wasn't a priority.

Reply:
Answering your question, you should use reward token decimals here because `supplySpeedMantissa` is in reward tokens. Not sure it's correct to call it `ratePerBlockMantissa` because rate feels like a _percentage_

Reply:
Your proposition with types makes sense to me, but if it involves heavy refactors I agree with you that's not a priority.

The ideal scenario would be to have the contacts fully documented with explanations on what the return types correspond to (I haven't had a chance to look at the new documentation thoroughly: maybe that's included in it already?).
Otherwise having the 3 rules you wrote somewhere in Notion would also be helpful.